app/src/main/java/fr/free/nrw/commons/auth/csrf/CsrfTokenClient.kt

```diff
@@ -22,6 +22,7 @@ class CsrfTokenClient(
 ) {
     private var retries = 0
     private var csrfTokenCall: Call<MwQueryResponse?>? = null
+    private val INVALID_TOKEN_ERROR_MESSAGE = "Invalid token, or login failure."
```

Comment:
I think you can make the constant public and use it from UploadWorker. The idea is to avoid repetition. To do so, you can add `const val INVALID_TOKEN_ERROR_MESSAGE = "Invalid token, or login failure."` to the companion object, and use from `UploadWorker` as `CsrfTokenClient.Companion.INVALID_TOKEN_ERROR_MESSAGE`. (Sorry if that was not clear.)

Is there a better way than using exception messages to communicate like this? (I feel like there is, but cannot pinpoint it right now.) If so, please do that by all means. If not, I think using constants like that is not too bad.

Reply:
I think this is a pretty straightforward approach to solving this, I would definitely think about other ways but for now fixing this (using constants)